plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2445,12 +2445,13 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         HiveTableHandle hiveTable = (HiveTableHandle) table;
 
         List<ColumnHandle> partitionColumns = ImmutableList.copyOf(hiveTable.getPartitionColumns());
-        List<HivePartition> partitions = partitionManager.getOrLoadPartitions(metastore, new HiveIdentity(session), hiveTable);
-
-        TupleDomain<ColumnHandle> predicate = createPredicate(partitionColumns, partitions);
 
+        TupleDomain<ColumnHandle> predicate = TupleDomain.all();
         Optional<DiscretePredicates> discretePredicates = Optional.empty();
-        if (!partitionColumns.isEmpty()) {
+        if (!partitionColumns.isEmpty() && hiveTable.getPartitions().isPresent()) {
```

Comment:
Did we decided that we don't want to list first N partitions (some small number) before bailing out?

Reply:
There is IMO no way to do that.
Also note that `getTableProperties` can be called multiple times during query life time.

Reply:
> Also note that getTableProperties can be called multiple times during query life time.

We cache results per query, so it shouldn't be a problem.

Which method is slow?
is it `HiveMetastore#getPartitionNamesByFilter` or `getPartitionsByNames`?

Reply:
Would getting all the "too many partitions" into the cache cause any memory related problems on the coordinator?